pkg/controller/garbagecollector/garbagecollector.go

```diff
@@ -375,9 +375,10 @@ func (gc *GarbageCollector) attemptToDeleteItem(item *node) error {
 	default:
 		// item doesn't have any solid owner, so it needs to be garbage
 		// collected. Also, none of item's owners is waiting for the deletion of
-		// the dependents, so GC deletes item with Default.
-		glog.V(2).Infof("delete object %s with Default", item.identity)
-		return gc.deleteObject(item.identity, nil)
+		// the dependents, so GC deletes item in Background.
+		glog.V(2).Infof("delete object %s in Background", item.identity)
+		policy := metav1.DeletePropagationBackground
+		return gc.deleteObject(item.identity, &policy)
```

Comment:
I think letting the presence of existing GC finalizers on `latest` affect the policy chosen here would make sense:

```go
var policy metav1.DeletionPropagation
switch {
case hasOrphanFinalizer(latest):
	// if an existing orphan finalizer is already on the object, honor it.
	policy = metav1.DeletePropagationOrphan
case hasDeleteDependentsFinalizer(latest):
	// if an existing foreground finalizer is already on the object, honor it.
	policy = metav1.DeletePropagationForeground
default:
	// otherwise, default to background.
	policy = metav1.DeletePropagationBackground
}
glog.V(2).Infof("delete object %s with propagation policy %s", item.identity, policy)
```

I'm less interested in the manual setting of a GC finalizer pre-deletion (which is not something I think we should encourage, or possibly even allow), and more interested in the following scenarios:

1. Orphaning deletion of X followed by cascading deletion of parent of X should honor the orphaning deletion, even if GC doesn't observe the orphaning delete before processing X as part of the cascade:
    ```
    // assume deployment foo <- replicaset foo-1 <- pods
    kubectl delete --cascade=false replicaset/foo-1 
    kubectl delete --cascade=true  deployment/foo
    ```

2. Foreground deletion of X followed by cascading deletion of parent of X should honor the foreground deletion and keep X around until the descendants of X are cleaned up, even if GC doesn't observe the foreground delete before processing X as part of the cascade